docs/hub/gguf-llamacpp.md

```diff
@@ -1,16 +1,46 @@
 # GGUF usage with llama.cpp
 
-Llama.cpp directly allows you to download and run inference on a GGUF simply by providing a path to the Hugging Face repo path and the file name. llama.cpp would download the model checkpoint in the directory you invoke it from:
+Llama.cpp allows you to download and run inference on a GGUF simply by providing a path to the Hugging Face repo path and the file name. llama.cpp would download the model checkpoint in the directory you invoke it from:
 
 ```bash
-./main \
+./llama-cli \
```

Comment:
Should we also add a small instruction for conversation mode `-cnv` ?

I think most users will prefer to use chat models.

Reply:
Good idea! Adding it in the snippet.

Reply:
I mean, the snippet should be split into 2 versions, because `-p` is used differently in 2 cases:

### Basic usage: run in conversation mode

```bash
./llama-cli
  --hf-repo lmstudio-community/Meta-Llama-3-8B-Instruct-GGUF \
  --hf-file Meta-Llama-3-8B-Instruct-Q8_0.gguf \
  -p "You are a helpful assistant" -cnv
```

### Advanced usage: run in completion mode

```bash
./llama-cli
  --hf-repo lmstudio-community/Meta-Llama-3-8B-Instruct-GGUF \
  --hf-file Meta-Llama-3-8B-Instruct-Q8_0.gguf \
  -p "I believe the meaning of life is " -n 128
```

Reply:
hmm. I think it might make it a bit too verbose for the doc. I can add a line just below it stating that you can run in `chat completion` mode, let me add it.

Reply:
Or maybe you can add only my first snippet?

When using `-cnv`, `-p` becomes system message, so it should be something like "You are helpful assistant" or "You are Bob". `-n` is also not relevant anymore in cnv mode, since the model will generate until it hits end-of-turn token.

Reply:
Fixed!